### PR TITLE
[13966] Documentation for content filter related allocation limits

### PR DESCRIPTION
**Pull request by MiguelCompany**
This PR adds documentation for the allocation limits introduced on eProsima/Fast-DDS#2552

### CODE REVIEW

File: docs/fastdds/dds_layer/core/policy/eprosimaExtensions.rst [hunk 1]
```diff
@@ -389,6 +393,35 @@ List of structure members:
...
+   * - |ContentFilterProperty::AllocationConfiguration::expression_parameters-api|
+     - :ref:`resourcelimitedcontainerconfig`
+     - ``{0, 100, 1}``
+
+* |ContentFilterProperty::AllocationConfiguration::expression_initial_size-api|:
+  Preallocated size of the filter expression.
+* |ContentFilterProperty::AllocationConfiguration::expression_parameters-api|:
+  Allocation configuration for the list of expression parameters.
+
 Example
```
Comment (JLBuenoLopez-eProsima) [inline docs/fastdds/dds_layer/core/policy/eprosimaExtensions.rst:425]: I would add a note in the example stating that the `content_filter` allocations cannot be set using XML for the moment.

File: docs/fastdds/use_cases/realtime/allocations.rst [hunk 2]
```diff
@@ -146,21 +146,23 @@ By default, a full dynamic behavior is used.
...
-so that it can be preallocated during the initialization of the DataWriter,
+and the collection of writer side content filters,
+so they can be preallocated during the initialization of the DataWriter,
 as shown in the example below.
 Please, refer to :ref:`resourcelimitedcontainerconfig` for a complete description of additional configuration
-alternatives given by this data member.
+alternatives given by these data members.
 
 
 +--------------------------------------------------------+
```
Comment (JLBuenoLopez-eProsima) [inline docs/fastdds/use_cases/realtime/allocations.rst:168]: Again, a note here stating that XML does not support the configuration of `reader_filters_allocations` should be included.

File: docs/fastdds/use_cases/realtime/allocations.rst [hunk 3]
```diff
@@ -146,21 +146,23 @@ By default, a full dynamic behavior is used.
 Parameters on the DataWriter
```
Comment (JLBuenoLopez-eProsima) [inline docs/fastdds/use_cases/realtime/allocations.rst:146]: I think this [section](https://fast-dds.docs.eprosima.com/en/latest/fastdds/use_cases/realtime/allocations.html#parameters-on-the-participant) should also include the information about `content_filter` allocations in the [Limiting the size of custom parameters](https://fast-dds.docs.eprosima.com/en/latest/fastdds/use_cases/realtime/allocations.html#limiting-the-size-of-custom-parameters)

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]

File: docs/fastdds/dds_layer/core/policy/eprosimaExtensions.rst (hunk 1 above)
Comment (MiguelCompany) [reply docs/fastdds/dds_layer/core/policy/eprosimaExtensions.rst:425]: Commit 4daf86a

File: docs/fastdds/use_cases/realtime/allocations.rst (hunk 2 above)
Comment (MiguelCompany) [reply docs/fastdds/use_cases/realtime/allocations.rst:168]: Commit ee4cca4

File: docs/fastdds/use_cases/realtime/allocations.rst (hunk 3 above)
Comment (MiguelCompany) [reply docs/fastdds/use_cases/realtime/allocations.rst:156]: Commit 1a58046

File: docs/fastdds/use_cases/realtime/allocations.rst [hunk 4]
```diff
@@ -116,6 +116,16 @@ for each participant and endpoint.
...
 
+|ParticipantResourceLimitsQos::content_filter-api| inside :ref:`participantresourcelimitsqos` provides members
+to configure the allocation behavior of content filter discovery information:
+
+* |ContentFilterProperty::AllocationConfiguration::expression_initial_size-api| sets the preallocated size of
+  the filter expression.
+* |ContentFilterProperty::AllocationConfiguration::expression_parameters-api| controls the allocation behavior
+  for the list of expression parameters. Refer to :ref:`resourcelimitedcontainerconfig` for a complete description
+  of the alternatives. Receiving information about a content filter with more parameters than the maximum
+  configured here, will make the filtering happen on the reader side.
```
Comment (JLBuenoLopez-eProsima) [inline docs/fastdds/use_cases/realtime/allocations.rst:127]: Please, break lines after each sentence.

**Review by JLBuenoLopez-eProsima — COMMENTED** [review]
`ResourceLimitedContainerConfig` structure is not included in the API reference and it would be nice to have if you do not mind adding it.
Also new class `ContentFilterProperty` is not included in the API reference.

Finally, the content filter information should also be part of the [full example](https://fast-dds.docs.eprosima.com/en/latest/fastdds/use_cases/realtime/allocations.html#full-example)

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM

**Pull request merged**